app/assets/javascripts/discourse/components/discourse-topic.js.es6

```diff
@@ -117,16 +135,11 @@ export default Ember.Component.extend(AddArchetypeClass, Scrolling, {
 
     this.set('hasScrolled', offset > 0);
 
+    // Trigger a header (topic title) update, but only if it is requried.
     const topic = this.get('topic');
     const showTopic = this.showTopicInHeader(topic, offset);
     if (showTopic !== this._lastShowTopic) {
-      this._lastShowTopic = showTopic;
-
-      if (showTopic) {
-        this.appEvents.trigger('header:show-topic', topic);
-      } else {
-        this.appEvents.trigger('header:hide-topic');
-      }
```

Comment:
I think some of the duplication here is fine, cause otherwise you have to call showTopicInHeader twice, for no great gain